**`Análisis exploratorio de los datos`**: _(Exploratory Data Analysis-EDA)_

Primero importamos las librerias que  vamos a necesitar

In [3]:
import pandas as pd
import matplotlib.pyplot as plt

Cargamos los archivos .csv a los que ya se les aplico las transformaciones

In [4]:
df_amazon_prime = pd.read_csv('../dataset_ETL/amazon_prime_ETL.csv', delimiter = ',',encoding = "utf-8")
df_disney_plus = pd.read_csv('../dataset_ETL/disney_plus_ETL.csv',delimiter= ',', encoding = "utf-8" )
df_hulu = pd.read_csv('../dataset_ETL/hulu_ETL.csv',delimiter= ',', encoding = "utf-8" )
df_netflix = pd.read_csv('../dataset_ETL/netflix_ETL.csv',delimiter= ',', encoding = "utf-8" )
df_ratings = pd.read_parquet('../dataset_ETL/ratings_ETL.csv')#, delimiter=',', encoding= "utf-8")

**Se corre un sample en cada dataframe para ver que tipo de datos hay**

In [5]:
df_amazon_prime.sample(20)

show_id     type                                           title  \
6195   s6196    movie                                 to catch a king   
6918   s6919    movie                                          dogged   
5065   s5066    movie                           sinbad: where u been?   
746     s747    movie                                  the benefactor   
3536   s3537    movie                     15-minute guided meditation   
6639   s6640    movie                                             ink   
4644   s4645    movie                                     premonition   
5622   s5623    movie                                     nazi undead   
4593   s4594    movie                                   crown heights   
7911   s7912    movie                                   out of bounds   
971     s972  tv show                                       small axe   
1045   s1046    movie                                          shadaa   
2066   s2067    movie                                 jackpot (tamil)   
8879   s8880    movie                                      golden job   
477     s478  tv show                 the private lives of the tudors   
769     s770  tv show  the amazing world of aviation series - 13-part   
1220   s1221    movie                                       rain fall   
88       s89    movie                                       winterset   
7483   s7484    movie              nick cannon: stand-up, don't shoot   
2130   s2131    movie                                      icy demons   

              director                                               cast  \
6195      clive donner             robert wagner, teri garr, horst janson   
6918  richard rowntree           sam saunders, nadia lamin, philip ridout   
5065      chuck vinson                                             sinbad   
746       andrew renzi           richard gere, dakota fanning, theo james   
3536       mark knight                                 anne-marie newland   
6639      laurent king  joseph malerba, julie chevallier, guillaume de...   
4644     gavin wilding    cynthia preston, adrian paul, christopher lloyd   
5622      steven spiel           georgia chara, leigh scully, andy mcphee   
4593       matt ruskin  ron canada, nestor carbonell, natalie paul, am...   
7911    richard tuggle  anthony michael hall, jenny wright, jeff kober...   
971                NaN  john boyega, letitia wright, shaun parkes, ama...   
1045     jagdeep singh         diljit dosanjh, neru bajwa, jagdeep sandhu   
2066        s. kalyaan                         jyotika, revathi, anandraj   
8879      kar lok chin               ekin cheng, jordon chan, michael tse   
477                NaN                                                NaN   
769                NaN                                                NaN   
1220        max mannix  gary oldman, kippei shiina, kyoko hasegawa, mi...   
88      alfred santell                burgess meredith, eduordo ciannelli   
7483         gil cunha                                        nick cannon   
2130               NaN                                                NaN   

            country date_added  release_year rating  \
6195            NaN        NaN          1984    16+   
6918            NaN        NaN          2017    18+   
5065            NaN        NaN          2010    16+   
746             NaN        NaN          2016    16+   
3536            NaN        NaN          2016    all   
6639            NaN        NaN          2018    16+   
4644            NaN        NaN          2000      r   
5622            NaN        NaN          2019    18+   
4593  united states        NaN          2017      r   
7911            NaN        NaN          1986      r   
971             NaN        NaN          2020    16+   
1045            NaN        NaN          2019    all   
2066            NaN        NaN          2019    all   
8879            NaN        NaN          2018    16+   
477             NaN        N

In [6]:
df_disney_plus.sample(20)

show_id     type                                              title  \
303     s304    movie                                          you again   
684     s685  tv show                         one day at disney (shorts)   
542     s543  tv show                            lost treasures of egypt   
916     s917    movie                                   going to the mat   
416     s417    movie                               d2: the mighty ducks   
1017   s1018    movie                                    life with mikey   
1391   s1392    movie                                   tuck everlasting   
1245   s1246    movie                                   the african lion   
350     s351    movie                 the nutcracker and the four realms   
144     s145    movie                                           garfield   
1010   s1011    movie  lego star wars: the new yoda chronicles – esca...   
760     s761    movie                                              brave   
44       s45    movie                               jingle all the way 2   
970     s971    movie                                          iron will   
1125   s1126  tv show                                   phineas and ferb   
178     s179    movie                             the croc that ate jaws   
455     s456  tv show                                       king fishers   
457     s458  tv show                                marvel funko shorts   
1449   s1450    movie              captain sparky vs. the flying saucers   
811     s812    movie                                      descendants 2   

                           director  \
303                    andy fickman   
684                             NaN   
542                             NaN   
916                  stuart gillard   
416                     sam weisman   
1017                   james lapine   
1391                    jay russell   
1245                    james algar   
350   lasse hallström, joe johnston   
144                    peter hewitt   
1010                 michael hegner   
760    mark andrews, brenda chapman   
44                        alex zamm   
970                    charles haid   
1125                            NaN   
178                   mark woodward   
455                             NaN   
457                             NaN   
1449                    mark waring   
811                    kenny ortega   

                                                   cast  \
303   kristen bell, sigourney weaver, jamie lee curt...   
684                                                 NaN   
542                                         rick robles   
916   andrew lawrence, khleo thomas, alessandra tore...   
416   emilio estevez, michael tucker, jan rubeš, kat...   
1017  michael fox, nathan lane, cyndi lauper, david ...   
1391  alexis bledel, ben kingsley, sissy spacek, amy...   
1245                                     winston hibler   
350   keira knightley, mackenzie foy, eugenio derbez...   
144   breckin meyer, jennifer love hewitt, stephen t...   
1010  tom kane, anthony daniels, matt sloan, trevor ...   
760   kelly macdonald, billy connolly, emma thompson...   
44    larry the cable guy, brian stepanek, santino m...   
970   mackenzie astin, kevin spacey, david ogden sti...   
1125  vincent martella, ashley tisdale, thomas sangs...   
178                                        dan nachtrab   
455                                      stephen mangan   
457                                                 NaN   
1449                                      charlie tahan   
811   dove cameron, cameron boyce, sofia carson, boo...   

                                        country  date_added  release_year  \
303                               united states  2021-02-12          2010   
684                                         NaN  2019-12-06          2019   
542                              united kingdom  2020-05-01          2018   
916                               united

In [7]:
df_hulu.sample(20)

show_id     type                                              title  \
2157   s2158    movie                              somm: into the bottle   
1034   s1035  tv show  is it wrong to try to pick up girls in a dungeon?   
1384   s1385  tv show                             if loving you is wrong   
1608   s1609  tv show          tyler perry's the haves and the have nots   
1606   s1607    movie                                    survival island   
2945   s2946    movie                                 sabrina down under   
8         s9  tv show                                             queens   
758     s759    movie                                   we were soldiers   
2815   s2816  tv show                                   jorge el curioso   
1191   s1192    movie                                   the sunlit night   
2332   s2333  tv show                                     deutschland 86   
963     s964    movie                           lost girls & love hotels   
2416   s2417  tv show                                               star   
2322   s2323    movie                               christmas perfection   
1532   s1533  tv show        jason vlogs smiles & smarts by pocket.watch   
1087   s1088    movie                                         pilgrimage   
808     s809    movie                                   a christmas hero   
542     s543    movie                                              stray   
2181   s2182    movie                                           das boot   
1663   s1664    movie                                            big men   

     director  cast                                 country  date_added  \
2157      NaN   NaN                           united states  2019-07-18   
1034      NaN   NaN                                   japan  2021-01-17   
1384      NaN   NaN                           united states  2020-09-23   
1608      NaN   NaN                                     NaN  2020-06-10   
1606      NaN   NaN                    united states, japan  2020-06-11   
2945      NaN   NaN                           united states  2014-04-25   
8         NaN   NaN                                     NaN  2021-10-20   
758       NaN   NaN                                     NaN  2021-05-01   
2815      NaN   NaN                                     NaN  2016-03-31   
1191      NaN   NaN                         germany, norway  2020-11-14   
2332      NaN   NaN                                 germany  2018-12-14   
963       NaN   NaN                                     NaN  2021-02-15   
2416      NaN   NaN                           united states  2018-09-27   
2322      NaN   NaN                           united states  2018-12-21   
1532      NaN   NaN                                     NaN  2020-07-03   
1087      NaN   NaN         ireland, belgium, united states  2020-12-30   
808       NaN   NaN                                     NaN  2021-04-08   
542       NaN   NaN                                     NaN  2021-07-01   
2181      NaN   NaN                 germany, czech republic  2019-06-17   
1663      NaN   NaN  united kingdom, denmark, united states  2020-05-20   

      release_year    rating                       listed_in  \
2157          2015    90 min                   documentaries   
1034          2015     tv-ma        action, adventure, anime   
1384          2014     tv-ma   black stories, drama, romance   
1608          2013     tv-14            black stories, drama   
1606          1996    96 min                   documentaries   
2945          1999      tv-g            comedy, family, teen   
8             2021     tv-14                    drama, music   
758           2002         r        action, adventure, drama   
2815          2006      tv-g         adventure, kids, latino   
1191          2020    82 min                 comedy, romance   
2332          2018     tv-ma  drama, international, thriller   
963           2020         r                  drama, romance   


In [8]:
df_netflix.sample(20)

show_id     type                                              title  \
7777   s7778  tv show                      power rangers super megaforce   
4532   s4533    movie                                         malevolent   
429     s430    movie                                     sanitation day   
4424   s4425    movie                          arjun: the warrior prince   
4596   s4597    movie                                        peace haven   
5035   s5036  tv show                          dragons: race to the edge   
7880   s7881    movie                                           rocky ii   
8615   s8616    movie                                            traitor   
5901   s5902    movie                                                tig   
8580   s8581    movie                                     thor: ragnarok   
4747   s4748  tv show                                       kath and kim   
1902   s1903    movie                                 all because of you   
489     s490    movie                                        back to q82   
4206   s4207  tv show                                             pocoyo   
4067   s4068    movie                                   quién te cantará   
2834   s2835    movie                                         i am jonas   
1274   s1275  tv show                                canine intervention   
5808   s5809    movie  jeff foxworthy and larry the cable guy: we’ve ...   
1662   s1663    movie                                           the suit   
7211   s7212    movie                                              kings   

                           director  \
7777                            NaN   
4532                  olaf de fleur   
429                   seyi babatope   
4424                arnab chaudhuri   
4596                    suman ghosh   
5035                            NaN   
7880             sylvester stallone   
8615             jeffrey nachmanoff   
5901  kristina goolsby, ashley york   
8580                  taika waititi   
4747                            NaN   
1902                     adrian teh   
489                  ahmed siddiqui   
4206                            NaN   
4067                  carlos vermut   
2834            christophe charrier   
1274                            NaN   
5808                      jay karas   
1662                  mohammad adel   
7211            deniz gamze ergüven   

                                                   cast  \
7777  andrew m. gray, ciara hanna, john mark louderm...   
4532  florence pugh, ben lloyd-hughes, celia imrie, ...   
429   blossom chukwujekwu, elozonam ogbolu, nse ikpe...   
4424  yudhveer bakoliya, ashok banthia, ravi khanwil...   
4596  soumitra chatterjee, paran banerjee, arun mukh...   
5035  jay baruchel, america ferrera, t.j. miller, ch...   
7880  sylvester stallone, talia shire, burt young, c...   
8615  don cheadle, guy pearce, saïd taghmaoui, neal ...   
5901                                         tig notaro   
8580  chris hemsworth, tom hiddleston, cate blanchet...   
4747  jane turner, gina riley, glenn robbins, peter ...   
1902  hairul azreen, janna nick, amerul affendi, hen...   
489   jassim al nabhan, abdel imam abdullah, abdulaz...   
4206                            stephen fry, alex marty   
4067  najwa nimri, eva llorach, carme elias, natalia...   
2834  félix maritaud, nicolas bauwens, tommy lee baï...   
1274                                      jas leverette   
5808                jeff foxworthy, larry the cable guy   
1662  tamer hosny, akram hosny, amina khalil, dalal ...   
7211  halle berry, daniel craig, lamar johnson, kaal...   

                                    country  date_added  release_year rating  \
7777                                    NaN  2016-01-01          2014  tv-y7   
4532                         united kingdom  2018-10-05          2018  tv-ma   
429                                 nigeria  2021-07-21          2020  tv-14   
4424                        

In [9]:
df_ratings.sample(20)

userid  scored  timestamp movieid  scoreds
8192460   256663     2.5 2008-02-26  ns4668      3.6
6247517    64461     2.5 2010-07-07  as1123      3.5
3413363    35501     3.5 2006-06-07  ns2570      3.6
8965174   265097     2.0 1997-05-30  hs1125      3.6
7045326    72675     2.0 2000-11-20  as5762      3.5
1266145    12979     5.0 2012-09-18  hs2373      3.6
3359173    34949     3.5 2005-04-03    ns28      3.4
10711969  121199     2.5 2008-01-11  ns7985      3.5
4265079    44246     4.5 2008-09-17  ns7199      3.5
5741260    59156     3.5 2008-07-28  as3109      3.6
4454139    45811     4.5 2016-02-14   ds596      3.6
5456581    56259     5.0 2015-10-25  ns2378      3.5
1473677    15149     5.0 2013-02-08  ns3890      3.6
2302515    23917     1.0 2000-08-10  as8751      3.6
1739697    18109     3.0 2004-01-12   as157      3.5
4282887    44469     2.5 2004-08-03  as5034      3.6
3508473    36544     4.0 2016-02-25  ns1050      3.6
7157216    73839     5.0 2013-11-28  ns4721      3.6
2139145    22223     5.0 2002-08-22  as4747      3.6
2886289    30070     2.5 2008-11-06  as6492      3.6

Observamos la informacion detallada de cada dataframe

In [10]:
df_amazon_prime.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9668 entries, 0 to 9667
Data columns (total 14 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   show_id        9668 non-null   object
 1   type           9668 non-null   object
 2   title          9668 non-null   object
 3   director       7586 non-null   object
 4   cast           8435 non-null   object
 5   country        672 non-null    object
 6   date_added     155 non-null    object
 7   release_year   9668 non-null   int64 
 8   rating         9668 non-null   object
 9   listed_in      9668 non-null   object
 10  description    9668 non-null   object
 11  id             9668 non-null   object
 12  duration_int   9668 non-null   int64 
 13  duration_type  9668 non-null   object
dtypes: int64(2), object(12)
memory usage: 1.0+ MB


In [11]:
df_disney_plus.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1450 entries, 0 to 1449
Data columns (total 14 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   show_id        1450 non-null   object
 1   type           1450 non-null   object
 2   title          1450 non-null   object
 3   director       977 non-null    object
 4   cast           1260 non-null   object
 5   country        1231 non-null   object
 6   date_added     1447 non-null   object
 7   release_year   1450 non-null   int64 
 8   rating         1450 non-null   object
 9   listed_in      1450 non-null   object
 10  description    1450 non-null   object
 11  id             1450 non-null   object
 12  duration_int   1450 non-null   int64 
 13  duration_type  1450 non-null   object
dtypes: int64(2), object(12)
memory usage: 158.7+ KB


In [12]:
df_hulu.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3073 entries, 0 to 3072
Data columns (total 14 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   show_id        3073 non-null   object 
 1   type           3073 non-null   object 
 2   title          3073 non-null   object 
 3   director       3 non-null      object 
 4   cast           0 non-null      float64
 5   country        1620 non-null   object 
 6   date_added     3045 non-null   object 
 7   release_year   3073 non-null   int64  
 8   rating         3073 non-null   object 
 9   listed_in      3073 non-null   object 
 10  description    3069 non-null   object 
 11  id             3073 non-null   object 
 12  duration_int   3073 non-null   int64  
 13  duration_type  2594 non-null   object 
dtypes: float64(1), int64(2), object(11)
memory usage: 336.2+ KB


In [13]:
df_netflix.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8807 entries, 0 to 8806
Data columns (total 14 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   show_id        8807 non-null   object
 1   type           8807 non-null   object
 2   title          8807 non-null   object
 3   director       6173 non-null   object
 4   cast           7982 non-null   object
 5   country        7976 non-null   object
 6   date_added     8797 non-null   object
 7   release_year   8807 non-null   int64 
 8   rating         8807 non-null   object
 9   listed_in      8807 non-null   object
 10  description    8807 non-null   object
 11  id             8807 non-null   object
 12  duration_int   8807 non-null   int64 
 13  duration_type  8804 non-null   object
dtypes: int64(2), object(12)
memory usage: 963.4+ KB


In [15]:
df_ratings.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11024289 entries, 0 to 11024288
Data columns (total 5 columns):
 #   Column     Dtype         
---  ------     -----         
 0   userid     int64         
 1   scored     float64       
 2   timestamp  datetime64[ns]
 3   movieid    object        
 4   scoreds    float64       
dtypes: datetime64[ns](1), float64(2), int64(1), object(1)
memory usage: 420.5+ MB


Se analizaran los valores faltantes

In [10]:
df_amazon_prime.isnull().sum()

show_id             0
type                0
title               0
director         2082
cast             1233
country          8996
date_added       9513
release_year        0
rating              0
listed_in           0
description         0
id                  0
duration_int        0
duration_type       0
dtype: int64

In [12]:
df_disney_plus.isnull().sum()

show_id            0
type               0
title              0
director         473
cast             190
country          219
date_added         3
release_year       0
rating             0
listed_in          0
description        0
id                 0
duration_int       0
duration_type      0
dtype: int64

In [11]:
df_hulu.isnull().sum()

show_id             0
type                0
title               0
director         3070
cast             3073
country          1453
date_added         28
release_year        0
rating              0
listed_in           0
description         4
id                  0
duration_int        0
duration_type     479
dtype: int64

In [13]:
df_netflix.isnull().sum()

show_id             0
type                0
title               0
director         2634
cast              825
country           831
date_added         10
release_year        0
rating              0
listed_in           0
description         0
id                  0
duration_int        0
duration_type       3
dtype: int64

In [14]:
df_ratings.isnull().sum()

userid       0
scored       0
timestamp    0
movieid      0
scoreds      0
dtype: int64

**Se analizaran las columnas con valores numericos**

Como a la columna 'date_added' no lo toma con formato DATETIME, se convierte.

In [14]:
df_amazon_prime['date_added'] = pd.to_datetime(df_amazon_prime['date_added'])
df_disney_plus['date_added'] = pd.to_datetime(df_disney_plus['date_added'])
df_hulu['date_added'] = pd.to_datetime(df_hulu['date_added'])
df_netflix['date_added'] = pd.to_datetime(df_netflix['date_added'])

In [12]:
df_amazon_prime['date_added'].min()

Timestamp('2021-03-30 00:00:00')

In [15]:
df_disney_plus['date_added'].min()

Timestamp('2019-10-01 00:00:00')

In [16]:
df_hulu['date_added'].min()

Timestamp('2006-07-14 00:00:00')

In [6]:
df_ratings['timestamp'].min()

Timestamp('1995-01-09 00:00:00')

**Se analizaran si hay duplicados**

In [4]:
df_ratings.duplicated().value_counts()

False    11013823
True        10466
dtype: int64

**Se identificaran valores atípicos en la columna "scored" del dataframe ratings**

In [8]:
q1 = df_ratings['scored'].quantile(0.25)
q3 = df_ratings['scored'].quantile(0.75)
iqr = q3 - q1

# define los límites para los outliers
lower_limit = q1 - 1.5 * iqr
upper_limit = q3 + 1.5 * iqr

# identifica los outliers
outliers = df_ratings[(df_ratings['scored'] < lower_limit) | (df_ratings['scored'] > upper_limit)]

# muestra los outliers
print(outliers)

          userid  scored  timestamp movieid  scoreds
0              1     1.0 2015-03-09   as680      3.5
24             1     0.5 2015-03-09  as7193      3.5
42             2     1.0 1997-06-23  hs1914      3.6
43             2     1.0 1997-06-23  as9455      3.5
84             4     1.0 2003-01-15  ns3279      3.6
...          ...     ...        ...     ...      ...
11024167  124373     1.0 2001-05-31  ns7039      3.5
11024171  124374     1.0 2001-11-28  ds1361      3.6
11024179  124374     1.0 2001-11-28  as9146      3.6
11024192  124375     1.0 1996-08-14   hs282      3.4
11024218  124376     1.0 2001-04-26  ns5214      3.6

[514764 rows x 5 columns]


El resultado que te devolvió es un subconjunto de datos de la tabla. En este subconjunto, se han eliminado todas las filas donde los valores de la columna "scored" se encuentran fuera del rango intercuartil, lo que sugiere que son posibles valores atípicos o "outliers". En otras palabras, estos son los datos que no son atípicos en la columna "scored". La tabla resultante tiene 514764 filas y 5 columnas. Cada fila contiene información de un usuario que ha calificado una película, incluyendo el ID del usuario, su calificación ("scored"), la fecha en que se hizo la calificación ("timestamp"), el ID de la película ("movieid") y el promedio de todas las calificaciones para esa película ("scored").

**CONCLUSIONES EDA**

GENERALES (considerando el objetivo y alcance de este proyecto):

- Las columnas duration tienen tanto información en minutos como de temporadas

- Las columnas cast tienen los actores separados por una coma

PARTICULARES

1- df_amazon_prime_ETL

- separdor ','+ encoding utf-8
- cararcteres especiales [ (30)
- valores faltantes
    + director   2082
    + cast       1233
    + country    8996
    + date_added 9513
- 9687 registros
- 12 columnas

2- df_disney_plus_ETL

- separdor ','+ encoding utf-8
- cararcteres especiales [ (NO)
- valores faltantes
    + director      473
    + cast          190
    + country       219
    + date_added    3
- 1450 registros
- 12 columnas


3- df_hulu_ETL

- separdor ','+ encoding utf-8
- cararcteres especiales [ (1)
- valores faltantes
    + director      3070
    + cast          3073
    + country       1453
    + date_added    28
    + description   4
    + duration_type 479
- 3090 registros
- 12 columnas

4- df_netflix_ETL

- valores faltantes
    + director 2634
    + cast 825
    + country 831
    + date_added 10
    + duration 3
- 8807 registros
- 12 columnas

5- df_rating_ETL
-valores faltantes 0
- 11024289 registros
- 5 columnas
- duplicados 10466

Al obtener el primer año de ingreso de la columna 'timestamp' (1995-01-09),se concluyo que es un valor fuera de lo normal dado que las plataformas abrieron años depues. 